### PR TITLE
Collision detector

### PR DESCRIPTION
**Pull request by JanSchollaert**
# Description

Improved the collision detector so it can detect if you are in a component that has isTrigger enabled. It also returns the objectId that you are about to or are colliding with.
Changed the physics system so you can move through those components as well.

# How can this be tested?
Test with this branch: [Weapon](https://github.com/Brick-Studios/BeastArena/tree/Weapon)
Run the game and try these things:

- Run in the weapon and press E (console should output "picked up weapon")
- Run in the black square and press E (console should output "this is not a weapon")
- Press E without standing in anything (console should output "you are not in an entity")
- You should not be able to walk through gorilla

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Code is const correct

### CODE REVIEW

**Comment by Peter-JanGootzen** [comment]
We should try to first merge #35 before this.

File: include/brickengine/collision_detector.hpp [hunk 1]
```diff
@@ -39,12 +39,14 @@ Inspector Collision Detector
...
+#include "brickengine/collision_return_values.hpp"
 
 class CollisionDetector {
 public:
 		CollisionDetector(std::shared_ptr<EntityManager> em);
-		//Returns amount of pixels that can still be moved to the collidable object.
-		double spaceLeft(int entity, Axis axis, Direction direction);
+		// Returns amount of pixels that can still be moved to the collidable object.
+		collisionReturnValues spaceLeft(int entity, Axis axis, Direction direction);
+    triggerReturnValues isInTrigger(int entity);
```
Comment (Peter-JanGootzen) [inline include/brickengine/collision_detector.hpp]: Wrong indentation

File: include/brickengine/collision_return_values.hpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+struct collisionReturnValues{
+    double spaceLeft;
+    int objectId;
+    collisionReturnValues(double spaceLeft, int objectId) : spaceLeft(spaceLeft), objectId(objectId){}
+};
+
+struct triggerReturnValues{
```
Comment (Peter-JanGootzen) [inline include/brickengine/collision_return_values.hpp]: Should have upercase Names, constructor needs to be on top and a space between the end of the initializers and _{}_

File: include/brickengine/components/colliders/rectangle_collider_component.hpp [hunk 3]
```diff
@@ -5,13 +5,14 @@
...
 public:
-    RectangleColliderComponent(double xScale, double yScale, double zScale);
+    RectangleColliderComponent(double xScale, double yScale, double zScale, bool isTrigger);
     static std::string getNameStatic();
 
     // data
     double xScale;
     double yScale;
     double zScale;
+    bool isTrigger;
```
Comment (Peter-JanGootzen) [inline include/brickengine/components/colliders/rectangle_collider_component.hpp]: Maybe add some documentation as this value name is quite vague if you are not familiar with Unity.

File: include/brickengine/collision_return_values.hpp [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    bool isInTrigger;
+    int objectId;
```
Comment (Peter-JanGootzen) [inline include/brickengine/collision_return_values.hpp]: I would suggest making this integer optional, to make the intent more clear that can be no object colliding right now.

File: include/brickengine/collision_return_values.hpp [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+struct CollisionReturnValues{
+    CollisionReturnValues(double spaceLeft, int objectId) : spaceLeft(spaceLeft), objectId(objectId) {}
+    double spaceLeft;
+    int objectId;
+};
+
+struct TriggerReturnValues{
+    TriggerReturnValues(bool isInTrigger, int objectId) : isInTrigger(isInTrigger), objectId(objectId) {}
+    bool isInTrigger;
+    std::optional<int> objectId;
```
Comment (Peter-JanGootzen) [inline include/brickengine/collision_return_values.hpp]: You don't actually use the optional rn, check [the reference](https://en.cppreference.com/w/cpp/utility/optional) on how to properly use it

**Comment by JanSchollaert** [comment]
Merged/implemented with parent-child branch

**Pull request closed without merging**